inverse_covariance/two_way_standard_scaler.py

```diff
@@ -0,0 +1,329 @@
+        #    the inverse transform which it sounds like we should not support.
+
+        # initialize variables on first pass
+        if self._initial_state():
+            self.col_mean_ = 0.
+            self.col_var_ = 0. if self.with_std else None
+            self.n_cols_seen_ = 0
+            self.row_mean_ = 0.
+            self.row_var_ = 0. if self.with_std else None
+            self.n_rows_seen_ = 0
+
+        self.col_mean_, self.col_var_, self.n_rows_seen_ = _incremental_mean_and_var(
+            X, self.col_mean_, self.col_var_, self.n_rows_seen_
+        )
+        self.row_mean_, self.row_var_, self.n_cols_seen_ = _incremental_mean_and_var(
+            X.T, self.row_mean_, self.row_var_, self.n_cols_seen_
+        )
+
+        self.row_scale_ = None
+        self.col_scale_ = None
+        if self.with_std:
+            self.row_scale_ = _handle_zeros_in_scale(np.sqrt(self.row_var_))
+            self.col_scale_ = _handle_zeros_in_scale(np.sqrt(self.col_var_))
+
+        return self
+
+    def transform(self, X, copy=False):
+        """Perform standardization by centering and scaling
+        Parameters
+        ----------
+        X : array-like, shape [n_rows, n_cols]
+            The data used to scale along the features axis.
+        copy : bool, optional (default: False)
+            Copy the input X or not.
+        """
+        check_is_fitted(self, "row_scale_")
+        X = check_array(
+            X,
+            accept_sparse=False,
+            copy=copy,
+            warn_on_dtype=True,
+            dtype=FLOAT_DTYPES,
+            estimator=self,
+        )
+        n_rows, n_cols = np.shape(X)
+        if self.n_cols_ != n_cols:
+            raise ValueError("Number of features must be same as for fit().")
+
+        if sparse.issparse(X):
+            raise NotImplementedError(
+                "Input is sparse: Algorithm for sparse matrices currently not supported."
+            )
+
+        return two_way_standardize(
+            X, with_mean=self.with_mean, with_std=self.with_std, copy=self.copy
+        )
+
+    def inverse_transform(self, X, copy=False):
+        """Scale back the data to the original representation
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data used to scale along the features axis.
+
+        copy : bool, optional (default: False)
+            Copy the input X or not.
+
+        Returns
+        -------
+        X_tr : array-like, shape [n_samples, n_features]
+            Transformed array.
+        """
+        check_is_fitted(self, "row_scale_")
+        if sparse.issparse(X):
+            raise NotImplementedError(
+                "Input is sparse: Algorithm for sparse matrices currently not supported."
+            )
+
+        warnings.warn("Reversing two way transformation is not accurate.")
+
+        # Q: Should ^ be a warning or should we just rais here and delete the
+        #    rest of the code?
+
```

Comment:
I've changed your exception to a warning and then the remaining code is the same as you had implemented.  Is it desirable to retain this method or should it be removed?

TODO: tests for this method.

Reply:
I don't understand the use case of inverse_transform really and only added what would be a reasonable one-step approximation to maintain consistency with API https://github.com/scikit-learn/scikit-learn/blob/f0ab589f/sklearn/preprocessing/data.py#L697. 

Is it preferable to just create a function that does nothing with a warning that this functionality is not supported? 

I think it is better to add approximate solution with some tests.

Reply:
Then I can leave the interface, raise so that it breaks but says why, and delete the code.

Reply:
I think its preferable not to provide an approximate solution if the utility is interface compatibility with a different utility-- it means maintenance for us and footguns for anyone using it.